Assets/Demo/DemoControls.cs

```diff
@@ -34,50 +473,45 @@ private void Initialize()
         m_menu_steamExitMenu = m_menu.GetAction("steamExitMenu");
         m_menu_submit = m_menu.GetAction("submit");
         m_menu_point = m_menu.GetAction("point");
-        m_Initialized = true;
     }
-
-    private void Uninitialize()
+    ~DemoControls()
     {
-        if (m_GameplayActionsCallbackInterface != null)
-        {
-            gameplay.SetCallbacks(null);
-        }
-        m_gameplay = null;
-        m_gameplay_fire = null;
-        m_gameplay_move = null;
-        m_gameplay_look = null;
-        m_gameplay_menu = null;
-        m_gameplay_steamEnterMenu = null;
-        if (m_MenuActionsCallbackInterface != null)
-        {
-            menu.SetCallbacks(null);
-        }
-        m_menu = null;
-        m_menu_navigate = null;
-        m_menu_click = null;
-        m_menu_steamExitMenu = null;
-        m_menu_submit = null;
-        m_menu_point = null;
-        m_Initialized = false;
+        UnityEngine.Object.Destroy(asset);
```

Comment:
Do we even need this?

Reply:
Since we generate the asset when creating the wrapper, I think we should destroy it. Otherwise it would eventually be cleaned up by `GarbageCollectSharedAssets` when loading a new scene, or when someone calls `Resources.UnloadUnusedAssets`, but I thought it would make sense to free it right away if we know we won't use it any more.